core/src/com/unciv/logic/civilization/diplomacy/DiplomacyFunctions.kt

```diff
@@ -113,6 +112,22 @@ class DiplomacyFunctions(val civInfo: Civilization){
                 ).toInt()
     }
 
+    fun canSignDefensivePact(): Boolean {
+        if (!civInfo.isMajorCiv()) return false
+        if (!civInfo.hasUnique(UniqueType.EnablesDefensivePacts)) return false
+        if (civInfo.gameInfo.ruleset.technologies.values
```

Comment:
This is a remnant from research agreement that is not relevant

Reply:
No, it's relevant to the research agreement, since that gives you techs that you can research
It's not relevant to defensive pact because that nothing to do with techs

And yes, that's the check :)